apps/photos/src/components/SingleInputForm.tsx

```diff
@@ -88,40 +90,85 @@ export default function SingleInputForm(props: SingleInputFormProps) {
+                        <Autocomplete
+                            id="free-solo-demo"
+                            filterSelectedOptions
+                            freeSolo
+                            options={props.optionsList.map((option) =>
+                                option.toString()
+                            )}
+                            onChange={(event, newValue) => {
+                                setFieldValue('inputValue', newValue);
+                            }}
+                            renderInput={(params) => (
+                                <TextField
+                                    {...params}
+                                    hiddenLabel={props.hiddenLabel}
+                                    variant="filled"
+                                    fullWidth
+                                    type={
+                                        showPassword ? 'text' : props.fieldType
                                     }
+                                    id={props.fieldType}
+                                    name={props.fieldType}
+                                    {...(props.hiddenLabel
+                                        ? { placeholder: props.placeholder }
+                                        : { label: props.placeholder })}
+                                    value={values.inputValue}
+                                    onChange={handleChange('inputValue')}
+                                    error={Boolean(errors.inputValue)}
+                                    helperText={errors.inputValue}
+                                    disabled={loading}
+                                    autoFocus={!props.disableAutoFocus}
+                                    autoComplete={props.autoComplete}
                                 />
-                            ),
-                        }}
-                    />
+                            )}
+                        />
+                    ) : (
+                        <TextField
+                            hiddenLabel={props.hiddenLabel}
+                            variant="filled"
+                            fullWidth
+                            type={showPassword ? 'text' : props.fieldType}
+                            id={props.fieldType}
+                            name={props.fieldType}
+                            {...(props.hiddenLabel
+                                ? { placeholder: props.placeholder }
+                                : { label: props.placeholder })}
+                            value={values.inputValue}
+                            onChange={handleChange('inputValue')}
+                            error={Boolean(errors.inputValue)}
+                            helperText={errors.inputValue}
+                            disabled={loading}
+                            autoFocus={!props.disableAutoFocus}
+                            autoComplete={props.autoComplete}
```

Comment:
Please check if we can put these common properties in a variable and
use
```
<TextField
{...commonParams}
>
```

Reply:
Okay, I will check